core/server_config.go

```diff
@@ -90,8 +126,22 @@ type HTTPConfig struct {
 	Address string `koanf:"address"`
 	// CORS holds the configuration for Cross Origin Resource Sharing.
 	CORS HTTPCORSConfig `koanf:"cors"`
+	// TLSMode specifies whether TLS is enabled for this interface, and which flavor.
+	TLSMode HTTPTLSMode `koanf:"tls"`
 }
 
+// HTTPTLSMode defines the values for TLS modes
+type HTTPTLSMode string
+
+const (
+	// DisabledHTTPTLSMode specifies that TLS is not enabled for this interface.
+	DisabledHTTPTLSMode HTTPTLSMode = "disabled"
+	// ServerCertTLSMode specifies that TLS is enabled for this interface, but no client certificate is required.
+	ServerCertTLSMode = "server-cert"
+	// MutualTLSMode specifies that TLS is enabled for this interface, and that it will require a client certificate.
+	MutualTLSMode = "server-and-client-cert"
```

Comment:
these are TLS options, so to me it makes more sense to describe the options in these terms. (disabled / tls / mutual-tls)

Reply:
I think "mTLS" is a bad term, since it implicates otherwise TLS is only used one way, correct term would be "TLS with mutual authentication", but that's probably too long. Will settle for your suggestion.

Reply:
It does look a bit weird, but I can live with this. 
Is authentication a better word for the first tls? so `authentication: disabled | tls | mutual-tls`.

Reply:
An upcoming change adds authentication, so then we'd get an `auth` field already. I'm not a big fan of my proposal either. Maybe just; `disabled`, `server`, `server-client`?